File: w08d04-Intro-to-Ruby/a1.js
const pizza = 'BBQ chicken Feast';
// console.log(pizza);

// const arr = [1,2,3,4,5,6,7,8,9,10];
// let i = 1;
// i++
// console.log(i);

// const printArr = arr => {
//     for (let elem of arr) {
//         console.log("The element is " + elem );
//     }
// }

// countChars ?

const countChars = str => {
    const result = {};
    console.log(result['h']);
    for (let c of str) {
        if (result[c])
            result[c]++;
        else
            result[c] = 1;
    }
    return result;
}
console.log(countChars('hello'));

File: w08d04-Intro-to-Ruby/a1.rb
Pizza = 'BBQ Chicken Feast'
require 'pry'

# puts Pizza
# there are alot of shorthands (faster way to write stuff)
# Arr = Array 1..10


# def printArr arr 
#     # code goes here....
    
#     arr.each do |elem|
#         puts "The element is #{elem.to_s}"
#     end
# end

# printArr Arr


# the equivalent of Objects in ruby is "hashes" or "hash"
# instead of keys, hashes have symbols, and symbols connect to values.

def count_chars(str)
    chars = str.split('')

    result = {}
    # puts result['h'.to_sym]
    # result['e'.to_sym] = 2
    # result[:e] = 2


    chars.each do |letter|

        # puts letter
        if result[letter.to_sym] == nil
            result[letter.to_sym] = 1
        else
            result[letter.to_sym] += 1
        end
    end
    yield

    return result
end

#  Yield example (small one)
puts count_chars('hello') { puts "Function has finished executing"}